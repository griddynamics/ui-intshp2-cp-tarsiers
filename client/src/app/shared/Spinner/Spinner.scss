@import '../../../styles/variables';
@import '../../../styles/mixin';

.fa-sun {
  @include abs-center;

  animation: spin 4s infinite linear;
  font-size: xx-large;
  color: $color-main;
}

.spin-position {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.spinner-container {
  position: relative;
  width: 100%;
  height: 100%;
}
