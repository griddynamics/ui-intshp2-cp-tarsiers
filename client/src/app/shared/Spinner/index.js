import React from 'react';
import './Spinner.scss';

const Spinner = props => {
  const { height } = props;
  const spinnerStyle = { height };

  return (
    <div className="spinner-container" style={spinnerStyle}>
      <div className="spin-position">
        <div className="spinner">
          <i className="fas fa-sun" />
        </div>
      </div>
    </div>
  );
};

export default Spinner;
