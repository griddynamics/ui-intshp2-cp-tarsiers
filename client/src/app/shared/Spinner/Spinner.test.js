import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './index';

describe('<Spinner />', () => {
  it('should match its snapshot', () => {
    const wrapper = shallow(<Spinner />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should renders without crashing', () => {
    expect(shallow(<Spinner />).length).toEqual(1);
  });
});
