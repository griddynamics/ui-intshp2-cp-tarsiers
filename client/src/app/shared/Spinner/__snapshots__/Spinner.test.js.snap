// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should match its snapshot 1`] = `
<div
  className="spinner-container"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <div
    className="spin-position"
  >
    <div
      className="spinner"
    >
      <i
        className="fas fa-sun"
      />
    </div>
  </div>
</div>
`;
